### PR TITLE
feat: rpc message based heartbeat

### PR DESCRIPTION
**Pull request by fengjiachun**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

_PLEASE DO NOT LEAVE THIS EMPTY !!!_

Please explain IN DETAIL what the changes are in this PR and why they are needed:

- Add `role` field to header, 0: datanode, 1: frontend
- Change the data type of the timestamp to int64.
- Add `MailboxMessage` to Heartbeat message for RPC

## Checklist

- [ ]  I have written the necessary comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: proto/greptime/v1/meta/common.proto [hunk 1]
```diff
@@ -10,6 +10,9 @@ message RequestHeader {
   uint64 cluster_id = 2;
   // member_id is the ID of the sender server.
   uint64 member_id = 3;
+  // 0: datanode
+  // 1: frontend
+  uint32 role = 4;
```
Comment (MichaelScofield) [inline proto/greptime/v1/meta/common.proto:15]: Can you use enum?

File: proto/greptime/v1/meta/heartbeat.proto [hunk 2]
```diff
@@ -92,3 +93,19 @@ message AskLeaderResponse {
...
+  uint64 region_id = 2;
+  // The following information plays a bigger role in making messages traceable and facilitating debugging.
+  string subject = 3;
+  string from = 4;
+  string to = 5;
+  // The unix timestamp in milliseconds.
+  int64 timestamp_millis = 6;
+
+  // The message body.
+  bytes payload = 7;
```
Comment (MichaelScofield) [inline proto/greptime/v1/meta/heartbeat.proto:110]: suggest adding serialization method like this:
```protobuf
oneof payload {
  bytes json = 7;
}
```

File: proto/greptime/v1/meta/heartbeat.proto [hunk 3]
```diff
@@ -92,3 +93,19 @@ message AskLeaderResponse {
 
   Peer leader = 2;
 }
+
+message MailboxMessage {
+  // The id is used to associate request and response.
+  uint64 id = 1;
+  // Each message belongs to only one region, which is specified by the region_id.
+  uint64 region_id = 2;
```
Comment (MichaelScofield) [inline proto/greptime/v1/meta/heartbeat.proto:101]:
```suggestion
```

**Review by MichaelScofield — APPROVED** [review]

**Review by Fengys123 — APPROVED** [review]
LGTM

**Pull request merged**